### PR TITLE
DRA: update testing pod yamls

### PR DESCRIPTION
**Pull request by bart0sh**
- added pod referencing external resource claim
- used ubuntu images for pod-external and pod-inline
- run simple test commands to check CDI injection
- set pod restart policy to Never

### CODE REVIEW

File: test/integration/cdi/example-driver/deploy/example/pod-external.yaml [hunk 1]
```diff
@@ -0,0 +1,40 @@
+# Two pods, one shared, global claim
```
Comment (pohly) [inline test/integration/cdi/example-driver/deploy/example/pod-external.yaml]: That comment is outdated, right?

File: test/integration/cdi/example-driver/deploy/example/pod-external.yaml [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+apiVersion: v1
+kind: Pod
+metadata:
+  name: test-external-claim
+spec:
+  restartPolicy: Never
+  containers:
+  - name: with-resource
+    image: ubuntu:22.04
+    command: ["sh", "-c", "ls -la /dev/tty8 ; set |grep envVar2"]
```
Comment (pohly) [inline test/integration/cdi/example-driver/deploy/example/pod-external.yaml]: What's the purpose of `ls -la /dev/tty8`?

Comment (pohly) [reply test/integration/cdi/example-driver/deploy/example/pod-external.yaml]: Why `envVar2`?

Comment (bart0sh) [reply test/integration/cdi/example-driver/deploy/example/pod-external.yaml]: it's set by POC kubelet plugin. Should we wait for it to be merged first?

File: test/integration/cdi/example-driver/deploy/example/pod-external.yaml (hunk 1 above)
Comment (bart0sh) [reply test/integration/cdi/example-driver/deploy/example/pod-external.yaml]: yep, fixed. thanks for pointing out

File: test/integration/cdi/example-driver/deploy/example/pod-external.yaml (hunk 2 above)
Comment (pohly) [reply test/integration/cdi/example-driver/deploy/example/pod-external.yaml]: The POC kubelet plugin shouldn't do that. Let's not merge this here.

Comment (bart0sh) [reply test/integration/cdi/example-driver/deploy/example/pod-external.yaml]: What it should do then? Which device[s] to inject, which env variables to set?

File: test/integration/cdi/example-driver/deploy/example/pod-external.yaml [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+---
+apiVersion: v1
+kind: Pod
+metadata:
+  name: test-external-claim
+spec:
+  restartPolicy: Never
+  containers:
+  - name: with-resource
+    image: ubuntu:22.04
```
Comment (pohly) [inline test/integration/cdi/example-driver/deploy/example/pod-external.yaml]: Pulling from Docker Hub is problematic. Can we use an image that is already used by Kubernetes E2E testing and hosted on gcr?

File: test/integration/cdi/example-driver/deploy/example/pod-external.yaml [hunk 4]
```diff
@@ -0,0 +1,41 @@
+# One external resource claim, one pod, two containers
+# One container uses resource, one does not
```
Comment (pohly) [inline test/integration/cdi/example-driver/deploy/example/pod-external.yaml]:
```suggestion
# One external resource claim, one pod, two containers.
# One container uses resource, one does not.
```

File: test/integration/cdi/example-driver/deploy/example/pod-inline.yaml [hunk 5]
```diff
@@ -1,22 +1,29 @@
+# One inline resource claim, one pod, two containers
+# One container uses resource, one does not
```
Comment (pohly) [inline test/integration/cdi/example-driver/deploy/example/pod-inline.yaml]:
```suggestion
# One inline resource claim, one pod, two containers.
# One container uses resource, one does not.
```

**Review by pohly — CHANGES_REQUESTED** [review]
Can you also updated shared-resource-claim.yaml accordingly?

File: test/integration/cdi/example-driver/deploy/example/pod-external.yaml (hunk 3 above)
Comment (bart0sh) [reply test/integration/cdi/example-driver/deploy/example/pod-external.yaml]: done

File: test/integration/cdi/example-driver/deploy/example/pod-external.yaml (hunk 2 above)
Comment (bart0sh) [reply test/integration/cdi/example-driver/deploy/example/pod-external.yaml]: resolved. discussed out of this PR

File: test/integration/cdi/example-driver/deploy/example/pod-external.yaml (hunk 3 above)
Comment (bart0sh) [reply test/integration/cdi/example-driver/deploy/example/pod-external.yaml]: @pohly btw, we can use smaller image, e.g. gcr.io/google-containers/busybox. Let me know if you think it's better.

Comment (pohly) [reply test/integration/cdi/example-driver/deploy/example/pod-external.yaml]: Let's use a busybox image that is already a dependency of E2E testing:
https://github.com/kubernetes/kubernetes/blob/305ad47627d7cc67d90389f5f8c27c37ebe4bf3a/test/utils/image/manifest.go#L240

-> registry.k8s.io/e2e-test-images/busybox:1.29-2

That will make it easier later on to integrate this with E2E testing.

Comment (bart0sh) [reply test/integration/cdi/example-driver/deploy/example/pod-external.yaml]: done, please review again

**Review by pohly — CHANGES_REQUESTED** [review]
Can you also update test/integration/cdi/example-driver/deploy/example/shared-resource-claim.yaml?

**Comment by bart0sh** [comment]
> Can you also update test/integration/cdi/example-driver/deploy/example/shared-resource-claim.yaml?

done. modified it and renamed to pod-shared.yaml for the name consistency.

**Pull request merged**